File: java/src/main/java/factory_method/product/UltraTruck.java
package factory_method.product;

import factory_method.product.part.Engine;
import factory_method.product.part.Tire;

public class UltraTruck extends Vehicle {
    public static final String NAME = "Ultra Truck";

    public UltraTruck() {
        name = NAME;
        engine = Engine.HYBRID;
        tire = Tire.OFFLOAD;
    }
}

File: src/main/kotlin/factory_method/product/UltraTruck.kt
package factory_method.product

import factory_method.product.part.Engine
import factory_method.product.part.Tire

class UltraTruck : Vehicle() {
    override var name = NAME
    override var engine: Engine = Engine.HYBRID
    override var tire: Tire = Tire.OFFLOAD

    companion object {
        const val NAME: String = "Ultra Truck"
    }
}